src/com/nutrons/framework/commands/Command.java

```diff
@@ -178,4 +177,23 @@ public Command killAfter(long delay, TimeUnit unit) {
     }
     return terms;
   }
+
+  public static PartialCommand beginWith(Runnable action) {
+    return new PartialCommand(action);
+  }
+
+  public static class PartialCommand {
```

Comment:
Add a convenience method to command that allows you to pass both the beginsWith and the endsWith at the same time.